File: Models/Section.cs
namespace StageursApp.Models;

public class Section
{
    public int? Id { get; set; }
    public string? Nom { get; set; }
    public int? FacultyId { get; set; }

    // Navigation
    public Faculty? Faculty { get; set; }
    public ICollection<Etudiant>? Etudiants { get; set; }
}

File: stageurs-laravel/app/Models/Section.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Section extends Model
{
    use HasFactory;

    protected $fillable = ['nom', 'faculty_id'];

    /**
     * Get the faculty that owns this section
     */
    public function faculty()
    {
        return $this->belongsTo(Faculty::class);
    }

    /**
     * Get the students in this section
     */
    public function etudiants()
    {
        return $this->hasMany(Etudiant::class);
    }
}
